src/main/kotlin/io/emeraldpay/dshackle/upstream/rpcclient/stream/JsonRpcStreamParser.kt

```diff
@@ -0,0 +1,243 @@
+                            whatCount.set(Count.CountQuotes)
+                            SingleResponse(processScalarValue(token, tokenStart, firstBytes, endStream), null)
+                        } else {
+                            when (token) {
+                                JsonToken.START_OBJECT -> {
+                                    whatCount.set(Count.CountObjectBrackets)
+                                    SingleResponse(
+                                        processAndCountBrackets(tokenStart, firstBytes, count, endStream, OBJECT_OPEN_BRACKET, OBJECT_CLOSE_BRACKET),
+                                        null,
+                                    )
+                                }
+                                JsonToken.START_ARRAY -> {
+                                    whatCount.set(Count.CountArrayBrackets)
+                                    SingleResponse(
+                                        processAndCountBrackets(tokenStart, firstBytes, count, endStream, ARRAY_OPEN_BRACKET, ARRAY_CLOSE_BRACKET),
+                                        null,
+                                    )
+                                }
+                                else -> {
+                                    throw IllegalStateException("'result' not an object nor array'")
+                                }
+                            }
+                        }
+                    } else if (parser.currentName == "error") {
+                        return SingleResponse(null, responseRpcParser.readError(parser))
+                    }
+                }
+            }
+            return null
+        }
+    }
+
+    private fun processAndCountBrackets(
+        tokenStart: Int,
+        bytes: ByteArray,
+        brackets: AtomicInteger,
+        endStream: AtomicBoolean,
+        openBracket: Byte,
+        closeBracket: Byte,
+    ): ByteArray {
+        for (i in tokenStart + 1 until bytes.size) {
+            countBrackets(bytes[i], brackets, openBracket, closeBracket)
+            if (brackets.get() == 0) {
+                endStream.set(true)
+                return Arrays.copyOfRange(bytes, tokenStart, i + 1)
+            }
+        }
+        return Arrays.copyOfRange(bytes, tokenStart, bytes.size)
+    }
+
+    private fun countBrackets(
+        byte: Byte,
+        brackets: AtomicInteger,
+        openBracket: Byte,
+        closeBracket: Byte,
+    ) {
+        if (byte == openBracket) {
+            brackets.incrementAndGet()
+        } else if (byte == closeBracket) {
+            brackets.decrementAndGet()
+        }
+    }
+
+    private fun processScalarValue(
+        token: JsonToken,
+        tokenStart: Int,
+        bytes: ByteArray,
+        endStream: AtomicBoolean,
+    ): ByteArray? {
+        if (token == JsonToken.VALUE_NULL) {
+            endStream.set(true)
+            return "null".toByteArray()
+        } else if (token == JsonToken.VALUE_STRING) {
```

Comment:
Looks like there are could be numeric or boolean value too.
Maybe add some log for default case?

Reply:
Added `else` branch for all others